caf: ble_adv: Update advertising data configuration documentation (#8457)
nrfconnect/sdk-nrf: doc/nrf/libraries/caf/ble_adv.rst

To enable the |ble_adv|, complete the following steps:

1. Enable and configure the :ref:`CAF Bluetooth LE state module <caf_ble_state>`.
#. Set the :kconfig:option:`CONFIG_CAF_BLE_ADV` Kconfig option.
#. Configure the advertising data and scan response data for undirected advertising.
   Both advertising data and scan response data are managed by Bluetooth LE advertising providers.
   See :ref:`bt_le_adv_prov_readme` for details.

>    See :ref:`nrf_desktop_ble_adv` for example of Bluetooth LE advertising providers configuration defined by an application.

Reviewer: ```suggestion
   Also see :ref:`nrf_desktop_ble_adv` for an example of the Bluetooth LE advertising providers configuration defined by an application.
```

Switching to slower advertising is done to reduce the energy consumption.

Power-down
==========

When the system goes to the Power-down state, the advertising stops.

Grace period
------------

> 

Reviewer: Hm, maybe keep line 116 about :kconfig:option:`CONFIG_CAF_BLE_ADV_SWIFT_PAIR_GRACE_PERIOD`? There is suddenly no info about what it does...

Author: kconfig:option:CONFIG_CAF_BLE_ADV_SWIFT_PAIR_GRACE_PERIOD was removed.
Now we introduce a generic grace period feature described below. Currently Swift Pair is only one of the use-cases for it.

Author: Currently we describe behavior here and then inform about the Kconfig option that is needed here.

Reviewer: OK, but then we would need at least one short line introducing the concept of the grace period, information that was present in the removed part: "the device advertises without the Swift Pair data for additional ... seconds to ensure that the user does not try to connect to the device that is no longer available."

> If an advertising data provider requests non-zero grace period, the advertising stop on power-down is delayed by the requested time.

Reviewer: ```suggestion
If an advertising data provider requests non-zero grace period with :kconfig:option:`CONFIG_CAF_BLE_ADV_SWIFT_PAIR_GRACE_PERIOD`, the stopping of advertising on power-down is delayed by the requested time.
```

Author: The Kconfig option does not belong to an advertising data provider.
Changed `advertising stop` to `stopping of advertising`

> Instead of instantly stopping advertising, the advertising enters grace period.

Reviewer: ```suggestion
Instead of instantly stopping, the advertising enters the grace period.
```

> During the grace period advertising is still active, but advertising data and scan response data can be modified.

Reviewer: ```suggestion
During the grace period, the advertising is still active, but advertising data and scan response data can be modified.
```

After the grace period ends, the advertising stops.

The grace period is requested for example by the `Swift Pair`_ advertising data provider (:kconfig:option:`CONFIG_BT_LE_ADV_PROV_SWIFT_PAIR`).

> During the grace period, Swift Pair data is removed from the advertising packet - device enters Swift Pair "cool-down" phase.

Reviewer: ```suggestion
During the grace period, Swift Pair data is removed from the advertising packet and the device enters Swift Pair's cool-down phase.
```

.. note::
   Make sure that :kconfig:option:`CONFIG_CAF_BLE_ADV_GRACE_PERIOD` Kconfig option is enabled if both following conditions are met:

>    * any of the providers requests grace period,

Reviewer: ```suggestion
   * Any of the providers requests the grace period.
```

   * :kconfig:option:`CONFIG_CAF_BLE_ADV_PM_EVENTS` is enabled.

>    The :kconfig:option:`CONFIG_CAF_BLE_ADV_GRACE_PERIOD` is enabled by default if Swift Pair advertising data provider is enabled in the configuration.

Reviewer: ```suggestion
   The :kconfig:option:`CONFIG_CAF_BLE_ADV_GRACE_PERIOD` is enabled by default if the Swift Pair advertising data provider is enabled in the configuration.
```

Switching to slower advertising is done to reduce the energy consumption.

Power-down
==========

> When the system goes to the power-down state, the advertising either instantly stops or enters grace period state.

Reviewer: ```suggestion
When the system goes to the power-down state, the advertising either instantly stops or enters the grace period state.
```
